### ISSUE TITLE
Add support for constructor property promotion

### ISSUE THREAD

**Issue by andrewnicols** [moodlehq/moodle-local_moodlecheck#111]
The following is valid in PHP from PHP 8.0:

```
<?php

class example
{

    /**
     * @param string $name
     */
    public function __construct(
        protected string $name,
    ) {
    }
}
```

But this is picked up as having incomplete and/or wrong parameters in the docblock.

**Comment by stronk7** [comment]
This is an interesting one, some months ago I was looking for information about how promoted properties should be documented:
```php
/**
 * @param string $name
 */
public function __construct(
    protected string $name,
) {
}
```

And / or:

```php
public function __construct(
    /** @var string $name */
    protected string $name,
) {
}
```

I imagine that the former (function PHPDoc block will be more compatible with tools out there, specifically documentation generators, so I would focus on that one... but I've seen requests to support also the later with Psalm, YouTrack...

Ciao :-)

**Comment by andrewnicols** [comment]
Already present, but confused by union types. Addressed in #113.
